### ISSUE TITLE
Feb Extension o²S²PARC Meta-Modeling Framework: Accept a Set of Data Samples and Iterate Over That Set

### ISSUE THREAD

**Issue by elisabettai** [ITISFoundation/osparc-issues#1111]
Description:
Extension of the o²S²PARC meta-modeling framework to accept a set of data samples and iterate over that set. Specifically, this will be required to sweep over a series of histological cross-sections.

Deliverable:
o²S²PARC functionality and demonstrator

Acceptance criteria:
A demonstrator exists that illustrates iterative pipeline execution over a set of data samples.

Out of scope: -

Deadline: Q2

[wrike Y7-MS 11.1.1.2: Extension o²S²PARC Meta-Modeling Framework: Accept a Set of Data Samples and Iterate Over That Set](https://www.wrike.com/open.htm?id=1162027237)

**Comment by elisabettai** [comment]
Esra should talk with Werner and Antonino. Werner to organize meeting.

**Comment by elisabettai** [comment]
Y7Q2 Status update (02/29/2024): Completed
Implementation: Iterative pipeline execution over a set of (image) data samples was demonstrated at the monthly update meeting on 02/28/2024.
The implementation makes use of a new Map Service that runs study clones in parallel over a set of parameters and returns the probed results (study output ports). It is leveraged as well for #1110 
As a use case, we leveraged a pipeline for automatic nerve cross-section segmentation and quantitative fiber population characterization (based on AxonDeepSeg, see #1093. We then used the extended metamodeling framework to apply that pipeline to a set of images from the SPARC Portal. Probability histograms and statistical descriptors are extracted from the morphometric results.
Only three lines of scripting are required to set up and execute the iterative pipeline execution. Importantly, the implementation ensures that the pipeline elaboration is completely independent from the general purpose (i.e., implementation agnostic) meta-modeling functionality.
